jon-nona/hack-assembler-typescript
# feat(file): read and write from file

# Pull Request

## Scope

HA1: read from an input file and write to an output file via an npm command

## Work Done

- write utils to parse input arguments and supply a default output file if none is supplied with tests
- add rxjs and chalk, ts-node, commands to assemble a file
- add chalk and output logging

## Steps to Test

- pull the branch, run npm install
- create a file on your desktop with some text in it
- run : `npm run assemble -- -i <path-to-your-input-assembly-file>` to run the assembler and output a .hack file. Optionally add a custom output file with `npm run assemble -- -i <path-to-your-input-assembly-file> -o <path-to-your-ouput-file>`
- check that the file was created with the same contents as the input file.
